SecurityPkg/Library/Tpm2DeviceLibDTpm/Tpm2Ptp.c

```diff
@@ -443,6 +443,11 @@ Tpm2GetPtpInterface (
     return Tpm2PtpInterfaceMax;
   }
 
+  if (Tpm2SvsmQueryTpmSendCmd ()) {

```

Comment:
I recommend using NULL class registration for SVSM TPM.
Please don't touch the current PTP for the real platform BIOS.

Reply:
Hi @jyao1. Thanks for your feedback and I apologize for the delay.

The API used to communicate with the SVSM vTPM is already under a NULL class. Could you elaborate a little bit more on what other NULL classes could be created?
- UefiCpuPkg/Library/AmdSvsmLibNull/AmdSvsmLibNull.c
- OvmfPkg/Library/AmdSvsmLib/AmdSvsmLib.c

Right, this patch series is failing the CI mostly because it is hooking into the PTP which is used by multiple platforms. The idea behind that was to register the SVSM vTPM as a platform driver, similar to what we are doing in the Linux kernel. If not the PTP, would you have any suggestion for places where we could add it? And could that also be enabled via "-D TPM2_ENABLE"?

Reply:
Thank you.

I will create the new SecurityPkg/Library/Tpm2InstanceLibSvsmVTpm and have it implement https://github.com/tianocore/edk2/blob/master/SecurityPkg/Include/Library/Tpm2DeviceLib.h similar to how Tpm2InstanceLibDTpm does.

It is not super clear though how I should register it in the Ovmf DSC file. I think I could do something similar to AmdSvsmLib.inf or TdxLib.inf. Would you have any suggestion/preference?

Reply:
ping @jyao1 ^^^